### DISCUSSION TITLE
Python: NF2FF mirroring, how to specify?

### DISCUSSION THREAD

**Discussion by VolkerMuehlhaus** [thliebig/openEMS-Project#267] [Q&A Usage] [answered]
Dear all,

I have an antenna model with PEC boundary at zmin, so I want to enable mirroring when creating the NF2FF box.

My code: 
`nf2ff_box = FDTD.CreateNF2FFBox(opt_resolution=[max_cellsize]*3,  mirror=[0,0,0,0,1,0])`

This gives an error message:
_nf2ff_box = FDTD.CreateNF2FFBox(opt_resolution=[max_cellsize]*3,  mirror=[0,0,0,0,1,0])
File "openEMS\openEMS.pyx", line 400, in openEMS.openEMS.openEMS.CreateNF2FFBox
TypeError: function() got multiple values for keyword argument 'mirror'_

How to specify this properly? From the documentation I understood that I need this array with 6 integers?
https://docs.openems.de/python/openEMS/nf2ff.html 

Any insight is much appreciated!
BR, Volker

**Comment by VolkerMuehlhaus** [comment]
Nobody?

The python documentation at 
https://docs.openems.de/python/openEMS/nf2ff.html
indicates that the mirror definition is an array for the 6 sides of the analysis box:

_mirror – (6,) int array – 0 (Off), 1 (PEC) or 2 (PMC) boundary mirroring_

However, the Matlab code at 
https://github.com/thliebig/openEMS/blob/7d7688a288bdbd16adc5e5d9cf9ac51ae75a5be7/matlab/CalcNF2FF.m
shows something very different !?

_% 'Mirror':  Add mirroring in a given direction (dir), with a given 
%            mirror type (PEC or PMC) and a mirror position in the given
%            direction.
%            Example: 'Mirror', {0, 'PMC', +100}_

If the Matlab call is correct, how to interpret the position value, is that using the unit set by mesh.SetDeltaUnit() or do I need to specify SI units here?

**Comment by thliebig** [comment]
Hello Volker, 

the python function CreateNF2FFBox setups mirror automatically by analyzing the boundary condiitons. I'm not sure if that is always desirable but this is the reason for the error . Maybe we have to change/fix that.

You could work around that by directly setting up the nf2ff object instead of using the CreateNF2FFBox "helper function". But this too is not ideal.

Or you could overwrite the mirror setup after creation (dirty hack):
`
nf2ff_box.mirror = [0,0,0,0,1,0]
`

But again if your boundary condition in zmin is already PEC, than this should already be set to 1?

**Reply by VolkerMuehlhaus** [reply]
Hi Thorsten, thanks for your feedback! My zmin boundary is PEC, and automatic evaluation should be fine.

So what value is required for the symmetry parameter then? The array from the documentation is obviously wrong, so what to use instead? Just one boolean to enable mirror?

Best regards
Volker

**Reply by thliebig** [reply, marked as answer]
The value is only 0 (off), 1 (PEC) or 2 (PMC) and the position is taken automatically from the simulation domain...
Is that what you mean?
